x/ibc/applications/transfer/types/codec.go

```diff
@@ -16,10 +22,18 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 }
 
 var (
-	// ModuleCdc references the global x/ibc-transfer module codec. Note, the codec
-	// should ONLY be used in certain instances of tests and for JSON encoding.
+	amino = codec.NewLegacyAmino()
+
+	// ModuleCdc references the global x/ibc transfer codec. Note, the codec should
+	// ONLY be used in certain instances of tests and for JSON encoding as Amino is
+	// still used for that purpose.
 	//
-	// The actual codec used for serialization should be provided to x/ibc-transfer and
+	// The actual codec used for serialization should be provided to x/ibc transfer and
 	// defined at the application level.
-	ModuleCdc = codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
+	ModuleCdc = codec.NewAminoCodec(amino)
```

Comment:
This breaks IBC. `ModuleCdc` should use Proto **only**. We need to create a separate module amino cdc to ensure that packet data bytes are proto encoded. Amino should only be used to register messages and marshal them for sign bytes